### ISSUE TITLE
Errors when 'Building the Webpack DLL....'

### ISSUE THREAD

**Issue by jleach** [mxstbr/react-boilerplate#946]
## Description

When running `npm install` the process works great until it reaches the Webpack DLL. Once this completes a number of parse warnings and errors scroll up the screen. See the attached console log for a sample. The errors don't seem to impact running my app; but I can't say I really know what the impact of them is.

I've created a Stackoverflow question on the subject but haven't received any good ideas to resolve the issue: http://stackoverflow.com/questions/39136933/how-do-i-fix-npm-install-errors-with-react-boilerplate?noredirect=1#comment65620189_39136933

## Steps to reproduce

Run `npm install` with the below mentions node version:

# Versions

$ npm -v
3.10.6
$ node -v
v4.5.0

[console.txt](https://github.com/mxstbr/react-boilerplate/files/450889/console.txt)

**Comment by gihrig** [comment]
This, from your log, looks to me like a successful dll build. 

> Building the Webpack DLL...
> Hash: a4147552ef2725f6b6f9
> Version: webpack 2.1.0-beta.15
> Time: 10698ms
>                       Asset     Size  Chunks             Chunk Names
> reactBoilerplateDeps.dll.js  8.65 MB       0  [emitted]  reactBoilerplateDeps
> chunk    {0} reactBoilerplateDeps.dll.js (reactBoilerplateDeps) 7.85 MB [rendered]
>  [1760] dll reactBoilerplateDeps 12 bytes {0} [built]
>      + 1765 hidden modules
> 

My guess is that the problem below is npm related:
> WARNING in ./~/sequelize/lib/dialects/mysql/connection-manager.js
> Module not found: Error: Can't resolve 'mysql'

Please confirm that `sequelize`, `mysql`, etc., are actually present in your `node_modules` folder and if not, try installing them individually with `npm install`.

It may help to:
1. Remove `node_modules`
2. `npm cache clean`
3. `npm install`

**Comment by mxstbr** [comment]
The `mysql` module is not meant to be used in the browser and will not work. Please use it on the backend, the server instead.

**Comment by jleach** [comment]
@mxstbr Thanks, I know it should not be used on the front end. Its in my `packages.json`file in the `dependencies` section. Only node server use it. For some reason its being picked up by `npm install` and something in react-boilerplate is processing it.

**Comment by jleach** [comment]
@gihrig I'm not actually using `mysql`, I'm using PostgreSQL `pg`. All the packages install fine and the app runs and works as expected. Its just something in the`npm install` phase has decided to do its own thing. Since the modules install fine I though it might be the post-install script(s) and the a part of react-boilerplate rather than npm itself.

**Comment by gihrig** [comment]
@jleach Interesting. As @mxstbr comments, since this is a front-end boilerplate, I don't see a need to resolve this issue, with any server-side package at least.

I would suggest that the design of your app should split the server and front-end into separate projects.

**Comment by mxstbr** [comment]
Have you tried adding the dependency to `internals/config.js` in the `exclude` array?

**Comment by nickgnd** [comment]
Hi, I had the same issue on `fs` module in my project. I've already answer on [SO](http://stackoverflow.com/a/39500648/4856698), but I report my solution below.

My error: 

```
❯ npm run build:dll

> crf@0.1.0 build:dll /Users/nico/_dev/crf
> node ./internals/scripts/dependencies.js

Building the Webpack DLL...
Hash: 9c3a018c9617c03c53cd
Version: webpack 2.1.0-beta.15
Time: 14427ms
                      Asset     Size  Chunks             Chunk Names
reactBoilerplateDeps.dll.js  3.32 MB       0  [emitted]  reactBoilerplateDeps
chunk    {0} reactBoilerplateDeps.dll.js (reactBoilerplateDeps) 2.96 MB [rendered]
  [925] dll reactBoilerplateDeps 12 bytes {0} [built]
     + 929 hidden modules

ERROR in ./~/dotenv/lib/main.js
Module not found: Error: Can't resolve 'fs' in '/Users/nico/_dev/crf/node_modules/dotenv/lib'
 @ ./~/dotenv/lib/main.js 3:9-22
```

Before all, I tried to insert `fs` dependency in `exclude` array in `internals/config.js` without success :(

So, to avoid the error during the Webpack DLL build, I update the `internals/webpack/webpack.dll.babel.js` file like this:

```
  // ...
  plugins: [
    new webpack.DllPlugin({ name: '[name]', path: join(outputPath, '[name].json') }), // eslint-disable-line no-new
  ],

  // mock fs
  node: {
    fs: 'empty',
  },
};
```

In webpack config the `node` key is specific to mock "various" node stuff (see [docs](https://github.com/webpack/docs/wiki/Configuration#node)). 
So, if I've understood well, with the snippet above I ignore the `fs` module for the DLL build which is not intended for the browser.

Hope it helps and thanks for the great job guys!

**Comment by prakash118** [comment]
In my case, i was using _request_ and I had the same issue. I fixed this issue by moving _request_ from _dependencies_ to _devDependencies_ . Worked for me. Also I think I should be able to fix it by adding the _request_ to the _exclude_ in package.json > _dllPlugin_ obj.

**Comment by bradleyy1012** [comment]
@mxstbr your solution of adding sequelize to the exclude array worked, thanks :D

**Comment by tuxrace** [comment]
@mxstbr I'm just wondering how did webpack recognizes if a package used for server side and not on the front-end?

**Comment by coleclayman** [comment]
I ended up trying @mxstbr 's solution of adding a list of `exclude` and also @nickgnd 's solution of adding a `node` option, but neither worked for me – I found another suggestion on a different GH issue and this is what ended up working for me.

In your `webpack.config.*.js` file, add the following to `plugins` after `new webpack.DllPlugin({})`:

`new webpack.IgnorePlugin(/your_package_name_here/)`. For me, it was yarn that was blowing stuff up.

Hope this helps!
